LucasPickering/slumber
# Initial support for openapiv3

## Description

Implementation of #106 .
Allows to import Slumber recipes from an OpenAPI specification.

## Known Risks

Not really applicable as this is a new feature, all stuff that is known to not work is clearly commented with a TODO.

Namely, it doesn't look like slumber supports : 
* Cookies
* Path parameters
* OAuth2
* OpenIDConnect

## QA

I added the classic `petstore` sample OpenAPI specification as a test. Feel free to import that and see if that works :)
There is one integration test that tests the feature as a whole, and then a few unit tests for the OpenAPI "reference resolving" module.   

## Checklist

- [x] Have you read `CONTRIBUTING.md` already?
  - Only user-facing changes belong in the changelog. Internal changes such as refactors should only be included if they'll impact users, e.g. via performance improvement.
- [x] Did you remove all TODOs?

-> I think I'd like to discuss a bit more before removing the TODOs and opening more issues, as these are not exactly small features :)
